src/main/java/com/tterrag/k9/commands/CommandClojure.java

```diff
@@ -403,20 +413,52 @@ public Object invoke(Object arg1) {
                     }
                     return TypeBindingPersistentMap.create(binding.bind("id", $ -> arg1), q);
                 }
-                
+
                 @Override
                 public String toString() {
                     return "Function:\n\t() -> list of quote IDs\n\tID -> Quote\n\n" + binding.toString();
                 }
             }
         ));
 
+        // A function for getting mapping data given a mapping channel
+        addContextVar("mappings", ctx -> Mono.just(new AFn() {
+            final TypeBinding<MappingInfo> mappingBinding = new TypeBinding<MappingInfo>("MappingInfo")
+                    .bind("channel", MappingInfo::getChannel)
+                    .bind("stable", MappingInfo::isStable)
+                    .bind("version", cmd -> cmd.getDownloader().getMappingsVersion(cmd.getMcVersion(), cmd.isStable()).block())
```

Comment:
Calling into the downloader is not a good idea, if this triggers a web request it will fail within trick context.

Reply:
How will it fail? I specifically tried to query stuff I did not have downloaded and it seemed to work fine